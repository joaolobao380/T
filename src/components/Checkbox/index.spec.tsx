import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import { useForm, FormProvider } from 'react-hook-form';

import { Checkbox } from '.'; // Ajuste o caminho conforme necessÃ¡rio

// Um componente wrapper para integrar o useForm
interface FormWrapperProps {
  label: string;
  name: string;
}

const FormWrapper = ({ label, name }: FormWrapperProps) => {
  const { control } = useForm({
    defaultValues: {
      [name]: null,
    },
  });
  return <Checkbox control={control} label="Teste label" name="checkbox" />;
};

describe('Checkbox Component', () => {
  const checkboxLabel = 'Teste label';

  it('Should renders correctly with the label', () => {
    const { getByText } = render(<FormWrapper name="testCheckbox" label={checkboxLabel} />);

    expect(getByText(checkboxLabel)).toBeTruthy();
  });

  it('Renders correctly Checkbox', () => {
    const checkboxLabel = 'Teste label';
    const { toJSON } = render(<FormWrapper name="testCheckbox" label={checkboxLabel} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
