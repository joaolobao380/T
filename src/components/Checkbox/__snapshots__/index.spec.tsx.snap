// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component Renders correctly Checkbox 1`] = `
<View
  testID="checkbox_testID"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
        "opacity": 1,
      }
    }
    testID="checkbox_testID_touchable"
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#444444",
            "borderRadius": 2,
            "borderWidth": 1,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        {
          "fontFamily": "Poppins-Regular",
        }
      }
    >
      Teste label
    </Text>
  </View>
</View>
`;
